tests/queries/0_stateless/03013_optimize_in_to_equal.sql

```diff
@@ -0,0 +1,16 @@
+DROP TABLE IF EXISTS test;
+CREATE TABLE test (x String, y Int32) ENGINE = MergeTree() ORDER BY x;
+INSERT INTO test VALUES ('a', 1), ('b', 2), ('c', 3);
+select * from test where x in ('a') SETTINGS allow_experimental_analyzer = 1;
+select '-------------------';
+select x in Null from test;
+select '-------------------';
+explain query tree select * from test where x in ('a') SETTINGS allow_experimental_analyzer = 1;
+select '-------------------';
+explain query tree select * from test where x in ('a','b') SETTINGS allow_experimental_analyzer = 1;
+select '-------------------';
+select * from test where x not in ('a') SETTINGS allow_experimental_analyzer = 1;
+select '-------------------';
+explain query tree select * from test where x not in ('a') SETTINGS allow_experimental_analyzer = 1;
+select '-------------------';
+select * from test where x not in (NULL) SETTINGS allow_experimental_analyzer = 1;
```

Comment:
also good to check `explain query` for this query and other cases, that do not support this conversion (x in array and x in tuple)

Reply:
done